Modules/_json.c

```diff
@@ -1667,12 +1726,15 @@ encoder_listencode_list(PyEncoderObject *s, _PyUnicodeWriter *writer,
         Py_CLEAR(ident);
     }
 
-    /* TODO: DOES NOT RUN
     if (s->indent != Py_None) {
-        indent_level -= 1;
+        indent_level--;
+        Py_DECREF(newline_indent);
+        Py_DECREF(separator_indent);
```

Comment:
These should be set to `NULL` to avoid double decrefs in case you bail.

```suggestion
        Py_CLEAR(newline_indent);
        Py_CLEAR(separator_indent);
```